### ISSUE TITLE
Keras installation issue -- Python ssl related

### ISSUE THREAD

**Issue by mgendarme** [rstudio/keras#1279]
Dear all,

I am experiencing some issues to install keras on windows.

My installation procedure using anaconda freshly installed (version 4.10.3):
```
conda create --name==rtf
conda activate rtf
conda install -c anaconda tensorflow==2.6.0
```

In R:
```
devtools::install_github("rstudio/keras")
library(keras)
install_keras(method = 'conda', envname = 'rtf')
```

My session info:
```
sessionInfo()
R version 4.1.1 (2021-08-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)  
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Germany.1252  LC_CTYPE=English_Germany.1252    LC_MONETARY=English_Germany.1252 LC_NUMERIC=C                     LC_TIME=English_Germany.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] keras_2.6.1.9000

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7            rstudioapi_0.13       whisker_0.4           magrittr_2.0.1        rappdirs_0.3.3        lattice_0.20-45       R6_2.5.1              rlang_0.4.11
 [9] tools_4.1.1           pkgbuild_1.2.0        grid_4.1.1            png_0.1-7             cli_3.0.1             withr_2.4.2           tfruns_1.5.0          remotes_2.4.1        
[17] rprojroot_2.0.2       crayon_1.4.1          tensorflow_2.6.0.9000 processx_3.5.2        Matrix_1.3-4          callr_3.7.0           base64enc_0.1-3       ps_1.6.0
[25] curl_4.3.2            zeallot_0.1.0         compiler_4.1.1        generics_0.1.0        prettyunits_1.1.1     reticulate_1.22       jsonlite_1.7.2
```

The error:
```
r$> install_keras(method = 'conda', envname = 'rtf')
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.

WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
Could not fetch URL https://pypi.org/simple/tensorflow/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/tensorflow/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
Requirement already up-to-date: tensorflow==2.6.* in c:\users\gendarme\anacon~2\envs\rtf\lib\site-packages (2.6.0)
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow-hub/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow-hub/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow-hub/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow-hub/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow-hub/
Could not fetch URL https://pypi.org/simple/tensorflow-hub/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/tensorflow-hub/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
ERROR: Could not find a version that satisfies the requirement tensorflow-hub (from versions: none)
ERROR: No matching distribution found for tensorflow-hub
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
Error: Error installing package(s): "\"tensorflow==2.6.*\"", "\"tensorflow-hub\"", "\"scipy\"", "\"requests\"", "\"Pillow\"", "\"h5py\"", "\"pandas\""
```

I have on purpose installed everything fresh, also tensorflow in Python in my conda environement runs fine.

Any clew how to finalize keras installation properly?

Many thanks in advance

**Comment by mgendarme** [comment]
I managed to install keras. I had to install fresh Rtools 4.0 and then reinstall reticulate:
`devtools::install_github("rstudio/reticulate")`
After that the installation of keras worked fine
`install_keras(method = 'conda', envname = 'rtf')`

However after install I am still experiencing issue. Python is not found for the environment keras and tensorflow are installed in:
Simple example:
```
x = imagenet_preprocess_input(x, mode = "caffe")
could not find a Python environment for C:/Users/Gendarme/Anaconda3/python.exe
```

I have posted as well on `reticulate` [1062](https://github.com/rstudio/reticulate/issues/1062)

If someone has experienced similar issue I am happy to hear about ways to fix that.

**Comment by mgendarme** [comment]
```
r$> Sys.setenv(RETICULATE_PYTHON = "C:\\Users\\Myself\\AppData\\Local\\r-miniconda\\envs\\r-reticulate\\python.exe")
    library(reticulate)
    py_config()
    import("tensorflow")
python:         C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate/python.exe
libpython:      C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate/python37.dll
pythonhome:     C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate
version:        3.7.11 (default, Jul 27 2021, 09:42:29) [MSC v.1916 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate/Lib/site-packages/numpy
numpy_version:  1.19.5

NOTE: Python version was forced by RETICULATE_PYTHON
2021-10-07 14:54:22.516375: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2021-10-07 14:54:22.516562: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
Module(tensorflow)
```

Trying the same with my anaconda rtf environement:
```
r$> py_path = "C:/Users/Myself/Anaconda3/envs/rtf/python.exe"
    Sys.setenv(RETICULATE_PYTHON = py_path)
    library(reticulate)
    py_config()
    import("tensorflow")
python:         C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate/python.exe
libpython:      C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate/python37.dll
pythonhome:     C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate
version:        3.7.11 (default, Jul 27 2021, 09:42:29) [MSC v.1916 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:/Users/Myself/AppData/Local/r-miniconda/envs/r-reticulate/Lib/site-packages/numpy
numpy_version:  1.19.5

NOTE: Python version was forced by RETICULATE_PYTHON
2021-10-07 14:56:59.119675: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2021-10-07 14:56:59.119857: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
Module(tensorflow)
```
I get the exatc same outcome. I can't force the use of the rtf environement. I am stuck to r-reticulate.

**Comment by t-kalinowski** [comment]
(I'm not an expert in conda or windows so take this with a grain of salt) 

Maybe when the `"rtf"` environment was created, it was done using the `r-miniconda/condabin/conda.exe` but that conda executable then got confused and installed the env under `"~/Anaconda3/envs"` instead of `"~/AppData/Local/r-miniconda/envs"`, and thus the python installation in `"C:/Users/Myself/Anaconda3/envs/rtf/"` is fundamentally broken and that's why reticulate can't bind to it. The fix would be to delete the environment (`unlink(""C:/Users/Myself/Anaconda3/envs/rtf", recursive = TRUE)`), and install again, this time giving an explicit path to `envname` and/or `conda` arguments to `install_keras()`, e.g. `install_keras(envname = "C:/Users/Myself/AppData/Local/r-miniconda/envs/rtf")`. 

Alternatively, if you want to go through your self-managed conda installation, you can open an anaconda prompt and run 
```bash
conda create --name rtf python=3.8
conda activate rtf
python -m pip install --ignore-installed tensorflow tensorflow-hub scipy requests Pillow h5py pandas 
```

Also, maybe call `normalizePath(py_path)` to convert `/` to `\` ?
